cpp/src/strings/search/contains_multiple.cu

```diff
@@ -0,0 +1,313 @@
+
+  auto const d_strings = column_device_view::create(input.parent(), stream);
+  auto const d_targets = column_device_view::create(targets.parent(), stream);
+
+  // copy the first byte of each target and sort them
+  auto first_bytes = rmm::device_uvector<u_char>(targets.size(), stream);
+  auto indices     = rmm::device_uvector<size_type>(targets.size(), stream);
+  {
+    auto tgt_itr = thrust::make_transform_iterator(
+      d_targets->begin<string_view>(),
+      cuda::proclaim_return_type<u_char>([] __device__(auto const& d_tgt) -> u_char {
+        return d_tgt.empty() ? u_char{0} : static_cast<u_char>(d_tgt.data()[0]);
+      }));
+    auto count_itr = thrust::make_counting_iterator<size_type>(0);
+    auto keys_out  = first_bytes.begin();
+    auto vals_out  = indices.begin();
+    auto num_items = targets.size();
+    auto cmp_op    = thrust::less();
+    auto sv        = stream.value();
+
+    std::size_t tmp_bytes = 0;
+    cub::DeviceMergeSort::SortPairsCopy(
+      nullptr, tmp_bytes, tgt_itr, count_itr, keys_out, vals_out, num_items, cmp_op, sv);
+    auto tmp_stg = rmm::device_buffer(tmp_bytes, stream);
+    cub::DeviceMergeSort::SortPairsCopy(
+      tmp_stg.data(), tmp_bytes, tgt_itr, count_itr, keys_out, vals_out, num_items, cmp_op, sv);
+  }
+
+  // remove duplicates to help speed up lower_bound
+  auto offsets = rmm::device_uvector<size_type>(targets.size(), stream);
+  thrust::sequence(rmm::exec_policy_nosync(stream), offsets.begin(), offsets.end());
+  auto const end = thrust::unique_by_key(
+    rmm::exec_policy_nosync(stream), first_bytes.begin(), first_bytes.end(), offsets.begin());
+  auto const unique_count =
+    static_cast<size_type>(thrust::distance(first_bytes.begin(), end.first));
+
+  // create output columns
+  auto const results_iter = cudf::detail::make_counting_transform_iterator(0, [&](int i) {
+    return make_numeric_column(data_type{type_id::BOOL8},
+                               input.size(),
+                               cudf::detail::copy_bitmask(input.parent(), stream, mr),
+                               input.null_count(),
+                               stream,
+                               mr);
+  });
+  auto results = std::vector<std::unique_ptr<column>>(results_iter, results_iter + targets.size());
+  auto d_results = [&] {
+    auto host_results_pointer_iter =
+      thrust::make_transform_iterator(results.begin(), [](auto const& results_column) {
+        return results_column->mutable_view().template data<bool>();
+      });
+    auto host_results_pointers =
+      std::vector<bool*>(host_results_pointer_iter, host_results_pointer_iter + results.size());
+    return cudf::detail::make_device_uvector_async(host_results_pointers, stream, mr);
+  }();
+
+  constexpr cudf::thread_index_type block_size = 256;
+  // calculated (benchmarked) for efficient use of shared-memory
+  constexpr size_type targets_threshold = 32;
```

Comment:
```suggestion
  constexpr size_type targets_threshold = cudf::detail::warp_size;
```

Reply:
The number of targets is independent of the warp-size. The number 32 is just a coincidence here. 
They should not be linked in this way.

Reply:
Thanks for the clarification.